DemoAppPush/NotificationService.swift

```diff
@@ -28,7 +28,11 @@ class NotificationService: UNNotificationServiceExtension {
                 return
             }
 
-            let localURL = URL(fileURLWithPath: path).appendingPathComponent(url.lastPathComponent)
+            // UNNotificationAttachment requires path extension to be set (fallback to jpeg if not set)
+            var localURL = URL(fileURLWithPath: path).appendingPathComponent(UUID().uuidString + url.lastPathComponent)
+            if localURL.pathExtension.isEmpty {
+                localURL.appendPathExtension("jpeg")
+            }
```

Comment:
Simplest fix to the missing path extension triggering UNNotificationAttachment failure. Not sure if we should put more effort into it. I tried `resourceValues` as well for downloaded URL but it fails in the extension.